core/src/main/scala/akka/persistence/r2dbc/internal/R2dbcExecutor.scala

```diff
@@ -138,74 +138,21 @@ class R2dbcExecutor(val connectionFactory: ConnectionFactory, log: Logger)(impli
   /**
    * Run DDL statement with auto commit.
    */
-  def executeDdl(logPrefix: String)(statement: Connection => Statement): Future[Done] = {
-    val startTime = System.nanoTime()
-
-    getConnection().flatMap { connection =>
-      connection.setAutoCommit(true).asFutureDone().flatMap { _ =>
-        val done =
-          try {
-            val stmt = statement(connection)
-            stmt.execute().asFuture().flatMap { result =>
-              result.getRowsUpdated.asFutureDone()
-            }
-          } catch {
-            case NonFatal(exc) =>
-              // thrown from statement function
-              Future.failed(exc)
-          }
-
-        done.failed.foreach { exc =>
-          log.debug("{} - DDL failed: {}", logPrefix, exc)
-          // auto-commit so nothing to rollback
-          connection.close().asFutureDone()
-        }
-
-        done.flatMap { _ =>
-          connection.close().asFutureDone().map { _ =>
-            if (log.isDebugEnabled())
-              log.debug("{} - DDL in [{}] µs", logPrefix, (System.nanoTime() - startTime) / 1000)
-            Done
-          }
-        }
+  def executeDdl(logPrefix: String)(statement: Connection => Statement): Future[Done] =
+    withAutoCommitConnection(logPrefix + " (DDL)") { connection =>
+      val stmt = statement(connection)
+      stmt.execute().asFuture().flatMap { result =>
+        result.getRowsUpdated.asFutureDone()
       }
     }
-  }
 
   /**
    * One update statement with auto commit.
    */
-  def updateOne(logPrefix: String)(statement: Connection => Statement): Future[Int] = {
-    val startTime = System.nanoTime()
-
-    getConnection().flatMap { connection =>
-      connection.setAutoCommit(true).asFutureDone().flatMap { _ =>
-        val rowsUpdated =
-          try {
-            val boundStmt = statement(connection)
-            updateOneInTx(boundStmt)
-          } catch {
-            case NonFatal(exc) =>
-              // thrown from statement function
-              Future.failed(exc)
-          }
-
-        rowsUpdated.failed.foreach { exc =>
-          log.debug("{} - Update failed: {}", logPrefix, exc)
-          // auto-commit so nothing to rollback
-          connection.close().asFutureDone()
-        }
-
-        rowsUpdated.flatMap { r =>
-          connection.close().asFutureDone().map { _ =>
-            if (log.isDebugEnabled())
-              log.debug("{} - Updated [{}] in [{}] µs", logPrefix, r, (System.nanoTime() - startTime) / 1000)
```

Comment:
There is not much different in terms of logging, so reusing new `withAutoCommitConnection`